File: backend/java-rules-service/src/main/java/com/karimerri/warroom/javaincident/api/dto/AddIncidentNoteRequest.java
package com.karimerri.warroom.javaincident.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;

@Schema(description = "Request used to add an investigation note")
public record AddIncidentNoteRequest(
        @Schema(example = "Karim")
        @NotBlank(message = "Author must not be blank")
        String author,

        @Schema(example = "Checked repository contract and found missing Optional handling.")
        @NotBlank(message = "Message must not be blank")
        String message
) {
}

File: frontend/flutter-rules-practice/lib/features/incidents/domain/add_incident_note_request.dart
class AddIncidentNoteRequest {
  final String author;
  final String message;

  const AddIncidentNoteRequest({
    required this.author,
    required this.message,
  });

  Map<String, dynamic> toJson() {
    return {
      'author': author,
      'message': message,
    };
  }
}